docs/devguide/xtypes.rst

```diff
@@ -821,6 +821,22 @@ Any enumerators that are not annotated take on the integer value one higher than
     @value(2) TWO
   };
 
+
+Marking member as optional
+==========================
+
+.. _xtypes--optional:
+
+@optional
+---------
+
+..
+    Sect<16.6.8.1>
+
+Applies to: struct members
+
+This annotation marks a member as optional.
```

Comment:
Please add some details on what this means for application developers in terms of the various IDL-to-language mappings.  That could be in the form of additional text here, or a link to other sections of the guide.  (Also add to NEWS?)

Reply:
This should address
1. What is an optional member?
2. What is required to use optional, i.e., the C++11 language mapping?
3. Possibly an example showing an IDL snippet declaring an optional member and a C++ snippet that shows how to set and safely access the optional value.